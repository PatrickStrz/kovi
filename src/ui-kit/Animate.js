import React, {Component} from 'react'
import PropTypes from 'prop-types'
import styled, {css} from 'styled-components'
import Transition from 'react-transition-group/Transition'
import TransitionGroup from 'react-transition-group/TransitionGroup'
// import {fadeOut, pulse} from 'styles/animations/keyframes'
const Box = styled.div`
  animation: ${props => props.animation};
  opacity: ${props => props.opacity};
  color: ${props => props.color}
`

class Animate extends Component {
  state = {
    in: false
  }
  componentDidMount = () => {
    this.props.inProp && this.setState({in: true})
  }
  componentWillReceiveProps = (nextProps) => {
    if (nextProps.inProp) {
      this.setState({in: true})
    }
    if (this.props.inProp && !nextProps.inProp){
      this.setState({in:false})
    }
  }

  // componentWillUnmount = () => {
  //   this.setState({in: false})
  // }
  getOpacity = (state) => {
    if (state === 'entering')   {
      return 0.5
    }
    if (state === 'exiting') {
      return 0.5
    }
  }

  render(){
    return(
      <Transition in={this.state.in} timeout={{enter:2000, exit:2000}} mountOnEnter={true} unmountOnExit={true}>
        {state => {
          return(
            <Box opacity={this.getOpacity(state)}> {this.props.child} {console.log(state)} </Box>
          )
        } }
      </Transition>
    )
  }
}

export default Animate
