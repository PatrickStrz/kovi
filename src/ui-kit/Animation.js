import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import {bounceInKeyframes} from 'styles/animations/keyframes'

const AnimationContainer = styled.div`
  animation: ${bounceInKeyframes} 0.5s;
`
const Animation = (props) => {
  return(
    <AnimationContainer>
      {props.children}
    </AnimationContainer>
  )
}

Animation.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Animation
