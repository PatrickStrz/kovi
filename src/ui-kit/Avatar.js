//react
import React from 'react'
import PropTypes from 'prop-types'
//other
import styled, {css} from 'styled-components'

//Profile avatar that displays inline
const Avatar = (props) => {
  return(
    <StyledAvatar size={props.size} imageUrl={props.imageUrl}/>
  )
}

Avatar.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
}

const StyledAvatar = styled.div`
  background-image: url(${props => props.imageUrl});
  ${props => `
      width: ${props.size};
      height: ${props.size};
    `}
  background-size: cover;
  /* center the image vertically and horizontally */
  background-position: center;
  border-radius: 50%;
  display: inline-block;
  ${props => css`
      min-width: ${props.size};
      max-width: ${props.size};
    `}
`

export default Avatar
