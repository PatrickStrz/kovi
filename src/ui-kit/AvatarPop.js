import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {Avatar} from 'ui-kit'
import {SelfDestruct} from 'ui-kit'

// user avatar that pops in and out on state change -->

class AvatarPop extends Component {
  state = {
    show: false,
  }
  static propTypes = {
    userPictureUrl: PropTypes.string.isRequired,
    uniqueEventId: PropTypes.string.isRequired,
    onShow: PropTypes.func,
    onHide: PropTypes.func,
    placeholder: PropTypes.node,
  }

  componentWillReceiveProps = (nextProps) => {
    if (nextProps.userPictureUrl &&
      (nextProps.uniqueEventId !== this.props.uniqueEventId)
    ){
      this.setState({show: true})
    }
  }

  componentWillUpdate = (nextProps, nextState) => {
    if (nextState.show && this.props.onShow){
      this.props.onShow()
    }
  }

  onSelfDestruct = () => {
    this.setState({show:false})
    this.props.onHide && this.props.onHide()
  }

  renderPoppingAvatar = () => {
    const {placeholder} = this.props
    const {userPictureUrl} = this.props

    if (this.state.show) {
      return(
        <SelfDestruct
          enterAnimationDuration={1500}
          exitAnimationDuration={1000}
          stayDuration={5000}
          onSelfDestruct={this.onSelfDestruct}
        >
          <Avatar imageUrl={userPictureUrl} size='25px'/>
        </SelfDestruct>
      )
    }
    else {
      if (placeholder) {
        return <div>{placeholder}</div>
      }
      else {
        return ''
      }
    }
  }

  render () {
    return (
      <div>
        {this.renderPoppingAvatar()}
      </div>
    )
  }
}

export default AvatarPop
