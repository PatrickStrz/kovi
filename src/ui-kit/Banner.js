import React,{Component} from 'react'
import PropTypes from 'prop-types'
//other
import styled from 'styled-components'
import {muiColors} from 'styles/theme/colors'
//components
import {FaIcon} from 'ui-kit/icons'

class Banner extends Component {
  static propTypes = {
      onExitClick: PropTypes.func.isRequired,
      text: PropTypes.node.isRequired,
  }

  render(){
    const {text, onExitClick} = this.props
    return(
      <Box>
        <Text>{text}</Text>
        <IconBox>
          <FaIcon
           faClassName="fa-close"
           color={muiColors.primary1}
           size="20px"
           onClick={onExitClick}
         />
        </IconBox>
      </Box>
    )
  }
}

const Box = styled.div`
  margin-bottom:25px;
  padding: 15px;
  position:relative;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${muiColors.primary100};
  border-radius: 3px;
`

const Text = styled.div`
  width: 80%;
  color: ${muiColors.primary1};
  text-align: left;
  line-height: 2;
`

const IconBox = styled.div`
  position:absolute;
  top:5px;
  right:5px;
`
export default Banner
