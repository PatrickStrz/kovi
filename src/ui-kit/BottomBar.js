import React, {Component} from 'react'
import PropTypes from 'prop-types'
//redux
import {connect} from 'react-redux'
import { bindActionCreators } from 'redux'
import {
  showNotificationsMobile,
  hideNotificationsMobile,
  showFilterMobile,
  hideFilterMobile,
  showCommunityMobile,
  hideCommunityMobile,
} from 'actions/bottombar-actions'
//helpers + other
import styled from 'styled-components'
//components
import {BottomNavigation, BottomNavigationItem} from 'material-ui/BottomNavigation'
import Paper from 'material-ui/Paper'
import FilterList from 'material-ui/svg-icons/content/filter-list'
import Notifications from 'material-ui/svg-icons/social/notifications'
import Public from 'material-ui/svg-icons/social/public'
import Dialog from 'ui-kit/Dialog'

const NotificationsIcon = <Notifications />
const FilterIcon = <FilterList />
const PublicIcon = <Public />

const NotificationsView = styled.h1`
  color: rgb(40, 125, 120);
`
const Filter = styled.h1`
  color: rgb(40, 125, 120);
`
const Community = styled.h1`
  color: rgb(40, 125, 120);
`

class BottomBar extends Component {

  static propTypes = {
     /* redux */
    showNotificationsMobile: PropTypes.func.isRequired,
    showFilterMobile: PropTypes.func.isRequired,
    showCommunityMobile: PropTypes.func.isRequired,
    isNotificationMobileOpen: PropTypes.bool.isRequired,
    isFilterMobileOpen: PropTypes.bool.isRequired,
    isCommunityMobileOpen: PropTypes.bool.isRequired,
    /* redux */
  }

  state = {
    selectedIndex: 0,
  }

  styles = {
    body: {
      width:'100vw'
    }
  }

  select = (index) => this.setState({selectedIndex: index})

  handleItemClick = (index, action) => {
    this.select(index) // Animates/ focuses on button
    action()
  }

  renderDialog = () => {
    const {
      isNotificationMobileOpen,
      isFilterMobileOpen,
      isCommunityMobileOpen,
      hideNotificationsMobile,
      hideFilterMobile,
      hideCommunityMobile,
    } = this.props

    let view
    let handleClose
    let title

    if (isNotificationMobileOpen){
      view = <NotificationsView>Notifications</NotificationsView>
      title = 'Notifications'
      handleClose = () => hideNotificationsMobile()
    }
    if (isCommunityMobileOpen){
      view = <Community>Community</Community>
      title = 'Community'
      handleClose = () => hideCommunityMobile()
    }
    if (isFilterMobileOpen){
      view = <Filter>Filter</Filter>
      title = "Filter"
      handleClose = () => hideFilterMobile()
    }
    return(
      view && (
      <Dialog
        isOpen={true}
        handleClose={handleClose}
        title={title}
      >
        {view}
      </Dialog>
      )
    )
  }

  render() {
    const {
      showNotificationsMobile,
      showFilterMobile,
      showCommunityMobile,
    } = this.props

    return (
      <Paper zDepth={1} style={this.styles.body}>
        <BottomNavigation selectedIndex={this.state.selectedIndex} >
          <BottomNavigationItem
            label="Filter"
            icon={FilterIcon}
            onTouchTap={()=> this.handleItemClick(0,showFilterMobile)}
          />
          <BottomNavigationItem
            label="Notifications"
            icon={NotificationsIcon}
            onTouchTap={()=> this.handleItemClick(1,showNotificationsMobile)}
          />
          <BottomNavigationItem
            label="Community"
            icon={PublicIcon}
            onTouchTap={()=> this.handleItemClick(2,showCommunityMobile)}
          />
        </BottomNavigation>
        {this.renderDialog()}
      </Paper>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
      showNotificationsMobile,
      hideNotificationsMobile,
      showFilterMobile,
      hideFilterMobile,
      showCommunityMobile,
      hideCommunityMobile,
    }, dispatch)
}

const mapStateToProps = (state) => ({
  isNotificationMobileOpen: state.app.bottomBar.isNotificationMobileOpen,
  isFilterMobileOpen: state.app.bottomBar.isFilterMobileOpen,
  isCommunityMobileOpen: state.app.bottomBar.isCommunityMobileOpen,
})

export default connect(mapStateToProps, mapDispatchToProps)(BottomBar)
