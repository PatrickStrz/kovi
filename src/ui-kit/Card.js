import React, {Component} from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
//helpers+other
import {colors} from 'styles/theme/colors'
import {CARD_Z_INDEX} from 'styles/z-index'
//components
import {Image} from 'ui-kit'

export default class Card extends Component{

  static propTypes = {
    imageUrl: PropTypes.string,
    text: PropTypes.string,
    onBodyClick: PropTypes.func,
    actions: PropTypes.node,
    highlight: PropTypes.bool,
    highlightColor: PropTypes.string,
    isLoading:  PropTypes.bool,
    backgroundColor: PropTypes.string,
  }

  render(){
    const {
      text,
      onBodyClick,
      actions,
      highlight,
      highlightColor,
      imageUrl,
      backgroundColor,
    } = this.props
    return(
      <CardBox
        highlight={highlight}
        highlightColor={highlightColor}
        backgroundColor={backgroundColor}
        >
        <CardBody>
          <ClickableBox onClick={onBodyClick}>
            <ImageBox onClick={onBodyClick}>
              <Image size="90px" url={imageUrl}/>
            </ImageBox>
            <Title>{text}</Title>
          </ClickableBox>
            {actions}
        </CardBody>
      </CardBox>
    )
  }
}

const CardBox = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  background-color: ${props =>{
    if (props.highlight){
      return props.highlightColor
    }
    else if (props.backgroundColor) {
      return props.backgroundColor
    }
    else return '#ffffff'
    }
  };
  width:100%;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  font-family: 'Open Sans', sans-serif;
  box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
  border-radius: 2px;
  z-index: ${CARD_Z_INDEX};
  margin-bottom: 5px;
  padding-right: 5px;
`

const CardBody = styled.div`
  width: 100%;
  margin-left: 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-right: 30px;
`
const ClickableBox = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  :hover{
    opacity: 0.5;
  }
`

const ImageBox = styled.div`
  margin: 10px;
`

const Title = styled.div`
  word-break: break-word;
  font-size: 16px;
  color: ${colors.grey};
  padding: 5px;
`
