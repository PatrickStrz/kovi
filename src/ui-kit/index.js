export {default as Comment} from './Comment'
export {default as Card} from './Card'
export {default as TextButton} from './TextButton'
export {default as InputWithProfile} from './InputWithProfile'
export {default as WarningDialog} from './WarningDialog'
export {default as Dialog} from './Dialog'
export {default as Avatar} from './Avatar'
export {default as AvatarPop} from './AvatarPop'
export {default as SelfDestruct} from './SelfDestruct'
export {default as MoreInfo} from './MoreInfo'
export {default as Popover} from './Popover'
export {default as ProgressMeter} from './ProgressMeter'
export {default as ImageUpload} from './ImageUpload'
export {default as Image} from './Image'
export {default as InputWithCharLimit} from './InputWithCharLimit'
export {default as MarkdownView} from './MarkdownView'
export {default as LoaderList} from './LoaderList'
export {default as Banner} from './Banner'
export {default as SingleTransition} from './SingleTransition'
