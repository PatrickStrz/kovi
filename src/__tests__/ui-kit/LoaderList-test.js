import React from 'react';
import {LoaderList} from 'ui-kit'
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const exampleLoader = <div>Loading...</div>
  const list = renderer.create(
    <LoaderList length={3} loader={exampleLoader} />
  ).toJSON();
  expect(list).toMatchSnapshot();
});
