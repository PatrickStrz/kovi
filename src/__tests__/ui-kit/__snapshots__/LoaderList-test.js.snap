exports[`test renders correctly 1`] = `
<div>
  <div
    className="sc-hzDkRC cBEVhi">
    <div>
      Loading...
    </div>
  </div>
  <div
    className="sc-hzDkRC cBEVhi">
    <div>
      Loading...
    </div>
  </div>
  <div
    className="sc-hzDkRC cBEVhi">
    <div>
      Loading...
    </div>
  </div>
</div>
`;
